bitcoin/src/psbt/mod.rs

```diff
@@ -427,6 +427,15 @@ impl Psbt {
                 k.get_key(&KeyRequest::Bip32(key_source.clone()), secp)
             {
                 secret_key
+            } else if let Ok(Some(sk)) = k.get_key(
+                &KeyRequest::Pubkey(xonly.public_key(secp256k1::Parity::Even).into()),
+                secp,
+            ) {
+                sk
+            } else if let Ok(Some(sk)) = k
+                .get_key(&KeyRequest::Pubkey(xonly.public_key(secp256k1::Parity::Odd).into()), secp)
+            {
```

Comment:
This is unfortunate. I wouldn't mind this in a backport but how about we modify `GetKey` trait to also have `get_by_x_only`?

I'd also prefer to provide `HashMap<XOnlyPublicKey, PrivateKey>` and do the retrieval by ignoring the parity (because both keys can be computed).

Reply:
Oh, yes, I forgot that `get_key` takes an enum (which is somewhat weird and possibly slightly slower than necessary but not really broken).

Still, I think having a map data structure that ignores parity when retrieving keys would make a lot of sense.